File: Lecture04/ListViews/app/src/main/java/com/codingblocks/listviews/models/Course.java
package com.codingblocks.listviews.models;

import java.util.ArrayList;
import java.util.Random;

/**
 * Created by arnav on 12/17/2017.
 */

public class Course {
    String name;
    String teacherName;
    int lectures;
    boolean onGoing;

    public Course(String name, String teacherName, int lectures, boolean onGoing) {
        this.name = name;
        this.teacherName = teacherName;
        this.lectures = lectures;
        this.onGoing = onGoing;
    }

    public String getName() {
        return name;
    }

    public String getTeacherName() {
        return teacherName;
    }

    public int getLectures() {
        return lectures;
    }

    public boolean isOnGoing() {
        return onGoing;
    }
    static final String[] sampleTeachers = {
            "Prateek", "Deepak", "Garima", "Rishab", "Arnav", "Aayush", "Harshit"
    };
    static final String[] sampleCourses = {
            "Pandora", "Elixir", "Launchpad", "Crux",
            "Perceptron", "Gamedev", "Competitive Coding", "Algo++"
    };
    public static ArrayList<Course> genRandCourses (int num) {
        ArrayList<Course> courses = new ArrayList<>(num);
        Random r = new Random();
        for (int i = 0; i < num; i++) {
            courses.add(new Course(
                    sampleCourses[r.nextInt(8)],
                    sampleTeachers[r.nextInt(7)],
                    15 + r.nextInt(10),
                    r.nextBoolean()
            ));
        }
        return  courses;
    }
}

File: Lecture06/KotlinSample/app/src/main/java/com/codingblocks/kotlinsample/models/Course.kt
package com.codingblocks.kotlinsample.models

import java.util.ArrayList

/**
 * Created by arnav on 12/24/2017.
 */
data class Course (
        val name: String,
        val teacherName: String,
        val lectures: Int,
        val onGoing: Boolean
)

fun genCourses (n: Int): ArrayList<Course> {
    val courses: ArrayList<Course> = ArrayList()
    for (i in 0 until n) {
        courses.add(
                Course(
                        "Android",
                        "Arnav",
                        22,
                        true
                )
        )
    }
    return  courses
}